agent/src/main/java/com/microsoft/applicationinsights/agent/internal/agent/AgentImplementation.java

```diff
@@ -153,6 +153,8 @@ public static String getAgentJarLocation() throws UnsupportedEncodingException {
         }
 
         String path = AgentImplementation.class.getProtectionDomain().getCodeSource().getLocation().getPath();
+        int index = path.lastIndexOf('/');
```

Comment:
Is it always in this notation across the platforms or sometimes can be '\\'?

Reply:
@Dmitry-Matveev It's a good question. I have an idea for 2 platforms I have tested so far. Java 8 has by default class loader of URLClassLoader type so it won't hit this part of the class .Java 9 has default classloader called "app" which is not of the type URLClassLoader and hence hits this part and only works for this change. I am not sure about other JVM's though.